Add matchTemplate nodelet

7675t opened the pull request:
- sample nodelet to use matchTemplate
- The template file is specified by parameter
- Test is needed

7675t committed:
Add matchTemplate nodelet

7675t committed:
Add matchTemplate nodelet

k-okada commented:
Thank you for contribution, btw did you develop this package on indigo or kinetic? if kinetic, did you used https://github.com/ros-perception/opencv_apps/pull/67 ?

7675t committed:
Remove imview at onInit

7675t commented:
I'm developing on Kinetic. And at first I used #67, however, it seems have a drawback that NODELET_INFO cannot output anything. So I don't apply #67 and not using debug_view parameter.

k-okada commented:
wow!!, that's very important information!! updated #67 so please check again when you have time.

7675t commented:
I confirmed #67 can show imview windows and print NODELET_INFO. Thank you!

k-okada committed:
matchTemplate: check opencv version, matchTemplate with mask is only suppreted >= 3.2

k-okada committed:
CMakeLists.txt move opencv_apps_add_nodlet location

k-okada committed:
match_template_nodelet.cpp, show error when iread fails

k-okada committed:
show template and result

k-okada committed:
match_template_nodelet.cpp: fix show rectanlge on result image

k-okada committed:
match_template_nodelet.cpp: publish and display matched value

k-okada committed:
match_template_nodelet.cpp: publish result rectangle

k-okada committed:
match_template_nodelet.cpp: enable to select mathing method from rqt_reconfigure

k-okada committed:
add test/test-match-template.test

7675t committed:
Add match_method dynamic parameter

7675t committed:
Add rostest for match_template

- Test using face_detector_withface_test_diamondback.bag
- Dynamic reconfigure for match_threshold
- Chnage to match only one in the scene image
- Change to use camera_info to publish gaze_vector

7675t committed:
Merge branch 'pr_from_534o' into add_matching_template

- Some PR #1 are adapted

7675t committed:
Remove default template and mask file name
